linux/hid.c

```diff
@@ -639,15 +690,19 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 
 		/* Get Report Descriptor Size */
 		res = ioctl(dev->device_handle, HIDIOCGRDESCSIZE, &desc_size);
-		if (res < 0)
-			perror("HIDIOCGRDESCSIZE");
-
+		if (res < 0) {
+			char errormsg[100];
+			snprintf(errormsg, sizeof(errormsg), "ioctl (GRDESCSIZE): %s", strerror(errno));
+			register_device_error(dev, errormsg);
+		}
 
 		/* Get Report Descriptor */
 		rpt_desc.size = desc_size;
 		res = ioctl(dev->device_handle, HIDIOCGRDESC, &rpt_desc);
 		if (res < 0) {
-			perror("HIDIOCGRDESC");
+			char errormsg[100];
+			snprintf(errormsg, sizeof(errormsg), "ioctl (GRDESC): %s", strerror(errno));
```

Comment:
I've taken the opportunity to also improve the messages here to match the ioctl messages further below :)